### PR TITLE
#163478818 Users should be able to highlight and comment on any text in an article

### PR DESCRIPTION
**Pull request by Ibidapo**
#### What does this PR do?
- Implement feature to highlight and comment on articles
 
#### Description of Task to be completed?
- Implement routes to create  and fetch highlights on articles
- Setup Controller functions to implement liking and disliking features
- Test feature for specified output
 
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/andela/denethor-ah-backend.git`
- Install Postgresql from [here](https://www.postgresql.org/download/)
- Create a database `authors-haven`
- Checkout to the branch `git checkout feature/163478818/user-highlight-article`
- Install application dependencies `npm install`
- Run the tests `npm test`

#### What are the relevant pivotal tracker stories?
- [Users should be able to highlight and comment on any text in an article](https://www.pivotaltracker.com/story/show/163478818)

### CODE REVIEW

File: server/api/controllers/article.js [hunk 1]
```diff
@@ -188,3 +196,77 @@ export const dislikeArticle = async (req, res) => {
...
+    const { params: { id: articleId }, user: { id: readerId } } = req;
+    const foundHighlights = await HighlightComment.findAll({
+      attributes: ['articleId', 'readerId', 'highlight', 'comment'],
+      where: {
+        articleId: { [Op.eq]: articleId },
+        readerId: { [Op.eq]: readerId }
+      }
+    });
+
+    if (foundHighlights.length === 0) {
```
Comment (ifiokeyo) [inline server/api/controllers/article.js]: @Ibidapo Please change line `250` to `if (!foundHighlights.length) {...}`.

File: server/api/helpers/tokenization/tokenize.js [hunk 2]
```diff
@@ -18,5 +18,9 @@ export const signToken = (payload, exp = '24h') => jwt.sign(payload, secret, { e
  * @function verifyToken
  * @param {Object} token - JWT token
  * @returns {string} Payload
+<<<<<<< HEAD
```
Comment (ifiokeyo) [inline server/api/helpers/tokenization/tokenize.js]: @Ibidapo Please remove all `merge conflict` indications where necessary.

File: server/api/helpers/encryption/encrypt.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import bcrypt from 'bcrypt';
```
Comment (ifiokeyo) [inline server/api/helpers/encryption/encrypt.js]: @Ibidapo Why do we have this file back in the project ?

Comment (Ibidapo) [reply server/api/helpers/encryption/encrypt.js]: The file must have been added while rebasing

File: server/api/controllers/article.js (hunk 1 above)
Comment (Ibidapo) [reply server/api/controllers/article.js]: Noted!

File: server/api/helpers/tokenization/tokenize.js (hunk 2 above)
Comment (Ibidapo) [reply server/api/helpers/tokenization/tokenize.js]: Noted!

**Pull request merged**